apex/provider/resource_mobility_groups_test.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://mozilla.org/MPL/2.0/
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package provider
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	. "github.com/bytedance/mockey"
+	"github.com/dell/terraform-provider-apex/apex/helper"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccMobilityGroupResource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: ProviderConfig + moblilityGroupResourceConfig,
+				Check:  resource.ComposeAggregateTestCheckFunc(),
+			},
+			{
+				Config: ProviderConfig + moblilityGroupResourceUpdateConfig,
+				Check:  resource.ComposeAggregateTestCheckFunc(),
+			},
+			{
+				Config:      ProviderConfig + moblilityGroupResourceErrorUpdateConfig,
+				ExpectError: regexp.MustCompile(`.*Error updating Mobility Group*.`),
+			},
+		},
+	})
+}
+
+func TestAccMobilityGroupError(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create Errors
+			{
+				PreConfig: func() {
+					FunctionMocker = Mock(helper.CreateMobilityGroup).Return(nil, nil, fmt.Errorf("Mock error")).Build()
+				},
+				Config:      ProviderConfig + moblilityGroupResourceConfig,
+				ExpectError: regexp.MustCompile(`.*Error creating Mobility Group*.`),
+			},
+			{
```

Comment:
@doriac11 can you add comment about what you are testing? even having separate function for each is fine to isolate the error.

Reply:
Yep I can add a comment on each of the error cases describing what we are exercising.